script.js

```diff
@@ -0,0 +1,179 @@
+const cityLocation = document.querySelector(".city");
+const countryLocation = document.querySelector(".country");
+const loader = document.querySelector(".loader");
+const container = document.querySelector(".weather-container");
+const disable = document.querySelector(".disable");
+const errorImg = document.querySelector(".img-fluiddd");
+const errorMessage = document.querySelector(".weather-content");
+const cards = document.querySelector(".cards");
+const dayNames = ["Sun", "Mon", "Tues", "Wed", "Thur", "Fri", "Sat"];
+let flag = true;
+const date = new Date();
+let cardContent = '';
+let spinner = false;
+const currentLocationWeather = () => {
+  if (navigator.geolocation) {
+    navigator.geolocation.getCurrentPosition(async function (position) {
+      const latitude = position.coords.latitude;
+      const longitude = position.coords.longitude;
+      const weatherDeatils = await fetchData(latitude, longitude);
+      weatherDeatils.forEach(element => {
+        displayWeatherInfo(element);
+      })
+
+    }, () => {
+      errorMessageHandler();
+      errorMessage.innerHTML = "PERMISSION BLOCKED";
+    });
+  }
+}
+
+currentLocationWeather();
+const handleInputData = async () => {
+  cardContent = "";
+  flag = true;
+  if (inputData.value.trim() == "") {
+    alert("Enter city name");
+    inputData.value = "";
+  }
+  else {
+    const res = await getLatitudeLongitude(inputData.value);
+    const lat = res?.lat
+    const lon = res?.lon
+    if (lat && lon) {
+      const weatherDeatils = await fetchData(lat, lon);
+      inputData.value = "";
+      weatherDeatils.forEach(element => {
+        displayWeatherInfo(element);
+      })
+    }
+    else {
+      errorMessageHandler();
+      errorMessage.innerHTML = "NOT FOUND";
+      document.body.style.backgroundImage = `url(/icons/bydefault.jpg)`;
+      inputData.value = "";
+    }
+  }
+}
+
+const getLatitudeLongitude = async (cityName) => {
+  try {
+    autoSpinner(true);
+    showCards(true);
+    const jsonData = await fatchDataFromUrl(`http://api.openweathermap.org/geo/1.0/direct?q=${cityName}&limit=5&appid=${key}`);
+    autoSpinner(false);
+    showCards(false);
+    return jsonData[0];
+  }
+  catch (e) {
+    console.log(e);
+  }
+}
+const fetchData = async (_latitude, _longitude) => {
+  try {
+    autoSpinner(true);
+    showCards(true);
+    const jsonResponse = await fatchDataFromUrl(`https://api.openweathermap.org/data/2.5/forecast?lat=${_latitude}&lon=${_longitude}&appid=${key}`);
+    autoSpinner(false);
+    showCards(false);
+    const cityName = jsonResponse.city.name;
+    const countryName = jsonResponse.city.country;
+    const fiveDaysForcaste = [];
+    const fiveDaysData = jsonResponse.list.filter((e) => {
+      const forcasteDate = new Date(e.dt_txt).getDate();
+      if (!(fiveDaysForcaste).includes(forcasteDate)) {
+        return fiveDaysForcaste.push(forcasteDate);
+
+      }
+    })
+    const forcastionDataObject = []
```

Comment:
we already have 5 days data then creating new array and storing same data doesn't makes any sense